cg/test/management/users.api.spec.js

```diff
@@ -154,3 +154,70 @@ describe('User password update', () => {
     expect((await res2.json()).jwt).toBeTruthy();
   });
 });
+
+describe('Listing management users', () => {
+  it('Should display an empty list when no manager is registered', async () => {
+    //Given
+    await db('users').del(); // It's fine to delete the contents of the managers table because no other table references it
+
+    //When
+    const res = await fetch('/api/management/users/list', {
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${managementToken}`
+      }
+    });
+
+    //Then
+    expect(res.ok).toBeTruthy();
+    expect(res.status).toBe(200);
+    expect(await res.json()).toEqual(expect.arrayContaining([]));
+  });
+
+  it('Should display correctly the registered managers', async () => {
+    //Given
+    await db('users').del(); // It's fine to delete the contents of the managers table because no other table references it
+    const managementToken1 = await managementJWT.sign({ id: '1' });
+    const managementToken2 = await managementJWT.sign({ id: '2' });
+    await fetch('/api/management/users/register', {
+      method: 'POST',
+      headers: { Authorization: `Bearer ${managementToken1}` },
+      body: {
+        username: 'manager1',
+        password: 'manager1_password'
+      }
+    });
+    await fetch('/api/management/users/register', {
+      method: 'POST',
+      headers: { Authorization: `Bearer ${managementToken2}` },
+      body: {
+        username: 'manager2',
+        password: 'manager2_password'
+      }
+    });
+
+    //When
+    const res = await fetch('/api/management/users/list', {
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${managementToken}`
+      }
+    });
+
+    //Then
+    expect(res.ok).toBeTruthy();
+    expect(res.status).toBe(200);
+    expect(await res.json()).toEqual(
+      expect.arrayContaining([
+        expect.objectContaining({
+          id: 5, // ID is 5 because 1 through 4 were deleted previously in the tests
```

Comment:
The ID's here are non deterministic. Unless you use the knex to fill the DB other tests can affect the ID numbering and break this test. Since you went for the route of calling register endpoint you should not care what IDs are being created and therefore you should change to:

```
id: expect.any(Number)
```

Reply:
Oks, what are other possibilities for this?